sql/core/src/main/scala/org/apache/spark/sql/scripting/SqlScriptingInterpreter.scala

```diff
@@ -44,9 +44,13 @@ case class SqlScriptingInterpreter(session: SparkSession) {
    */
   def buildExecutionPlan(
       compound: CompoundBody,
-      args: Map[String, Expression]): Iterator[CompoundStatementExec] = {
-    transformTreeIntoExecutable(compound, args)
-      .asInstanceOf[CompoundBodyExec].getTreeIterator
+      args: Map[String, Expression],
+      context: SqlScriptingExecutionContext): Iterator[CompoundStatementExec] = {
+    val compoundBodyExec = transformTreeIntoExecutable(compound, args, context)
+      .asInstanceOf[CompoundBodyExec]
+    context.frames.addOne(new SqlScriptingExecutionFrame(compoundBodyExec.getTreeIterator))
+    compoundBodyExec.enterScope()
```

Comment:
is this line what we talked about removing from here?

Reply:
Yes, context initialization is moved from here.